Source/WorkersES6/decodeGoogleEarthEnterprisePacket.js

```diff
@@ -201,15 +201,29 @@ function processMetadata(buffer, totalSize, quadKey) {
 function processTerrain(buffer, totalSize, transferableObjects) {
   var dv = new DataView(buffer);
 
+  // Each tile is split into 4 parts.
+  var advanceTile = function (pos) {
+    for (var quad = 0; quad < 4; ++quad) {
+      // Guard against reading past the end of buffer. Partial tiles are not
+      // allowed.
+      var size = dv.getUint32(pos, true);
+      if (size > totalSize) {
```

Comment:
Wouldn't this be 
```
if ((pos+sizeOfUint32+size) > totalSize) {
```
because `size` is just the size of the quad and will always be less than `totalSize`.

Reply:
Actually size is either a valid number of bytes for the tile subpart (quad) when the buffer still has tiles left to be parsed out, or it will be something else completely and this can easily be found because it is not a small number. The extra bytes in the terrain packet buffer are used for something else completely and not relevant to the CesiumJS viewer. Perhaps making the test as you suggest is slightly more correct and I will try that out.

Reply:
So after more digging in the legacy Google Earth client code, I have learned that the terrain packet will always have 5 tiles, and each tile will always have 4 sub-meshes. Any other combination means something went wrong and the parsing should be aborted. The remaining bytes after the 20 sub-meshes will be water surface meshes, which appear to be unused by CesiumJS so should be ignored. I will rework the parsing with this understanding.